serviceregistry-server/src/main/java/no/difi/meldingsutveksling/serviceregistry/controller/ServiceRecordController.java

```diff
@@ -42,159 +52,160 @@
+        if (!optionalEntityInfo.isPresent()) {
+            return notFoundResponse(String.format("Entity with identifier '%s' not found.", identifier));
+        }
+        Optional<Process> optionalProcess = processService.findByIdentifier(processIdentifier);
+        if (!optionalProcess.isPresent()) {
+            return notFoundResponse(String.format("Process with identifier '%s' not found.", processIdentifier));
+        }
+        Entity entity = new Entity();
+        EntityInfo entityInfo = optionalEntityInfo.get();
+        entity.setInfoRecord(entityInfo);
+        ServiceRecord serviceRecord = null;
+        ProcessCategory processCategory = optionalProcess.get().getCategory();
+        if (processCategory.equals(ProcessCategory.DIGITALPOST) && shouldCreateServiceRecordForCitizen().test(entityInfo)) {
+            if (clientOrgnr == null) {
+                return errorResponse(HttpStatus.UNAUTHORIZED, "No authentication provided.");
+            }
+            entity.getServiceRecords().addAll(serviceRecordFactory.createDigitalpostServiceRecords(identifier, auth, clientOrgnr));
+        }
+        if (processCategory == ProcessCategory.ARKIVMELDING) {
+            Optional<ServiceRecord> osr = serviceRecordFactory.createArkivmeldingServiceRecord(identifier, processIdentifier, securityLevel);
+            if (osr.isPresent()) {
+                serviceRecord = osr.get();
+            }
+            if (serviceRecord == null) {
+                return notFoundResponse(String.format("Arkivmelding process '%s' not found for receiver '%s'.", processIdentifier, identifier));
+            }
+        }
+        if (processCategory == ProcessCategory.EINNSYN) {
+            Optional<ServiceRecord> dpeServiceRecord = serviceRecordFactory.createEinnsynServiceRecord(identifier, processIdentifier);
+            if (!dpeServiceRecord.isPresent()) {
+                return notFoundResponse(String.format("eInnsyn process '%s' not found for receiver '%s'.", processIdentifier, identifier));
+            }
+            serviceRecord = dpeServiceRecord.get();
+        }
+        entity.getServiceRecords().add(serviceRecord);
+        return new ResponseEntity<>(entity, HttpStatus.OK);
+    }
+
     /**
      * Used to retrieve information needed to send a message to an entity (organization or a person) having the provided
      * identifier
      *
      * @param identifier of the organization/person to receive a message
-     * @param obligation determines service record based on the recipient being notifiable
      * @return JSON object with information needed to send a message
      */
-    @RequestMapping(value = "/identifier/{identifier}", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    @GetMapping(value = "/identifier/{identifier}", produces = MediaType.APPLICATION_JSON_VALUE)
     @ResponseBody
     @SuppressWarnings("squid:S2583")
     public ResponseEntity entity(
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to throw at most one checked exception instead of: no.difi.meldingsutveksling.serviceregistry.exceptions.SecurityLevelNotFoundException, no.difi.meldingsutveksling.serviceregistry.krr.KRRClientException, no.difi.meldingsutveksling.serviceregistry.CertificateNotFoundException, no.difi.meldingsutveksling.serviceregistry.krr.DsfLookupException, no.difi.meldingsutveksling.serviceregistry.service.brreg.BrregNotFoundException [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1160)